tools/benchmark/src/main/java/com/datadog/benchmark/internal/SpanEventSerializer.kt

```diff
@@ -0,0 +1,87 @@
+/*
+ * Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
+ * This product includes software developed at Datadog (https://www.datadoghq.com/).
+ * Copyright 2016-Present Datadog, Inc.
+ */
+
+package com.datadog.benchmark.internal
+
+import com.datadog.android.core.constraints.DataConstraints
+import com.datadog.android.core.constraints.DatadogDataConstraints
+import com.datadog.android.core.internal.utils.NULL_MAP_VALUE
+import com.datadog.benchmark.internal.model.SpanEvent
+import com.google.gson.JsonArray
+import com.google.gson.JsonObject
+import com.google.gson.JsonPrimitive
+import java.util.Date
+
+internal class SpanEventSerializer(
+    private val dataConstraints: DataConstraints = DatadogDataConstraints()
+) {
+
+    // region Serializer
+
+    fun serialize(env: String, spanEvents: List<SpanEvent>): String {
+        val spans = JsonArray(1)
+        spanEvents.forEach {
+            spans.add(sanitizeKeys(it).toJson())
+        }
+        val jsonObject = JsonObject()
+        jsonObject.add(TAG_SPANS, spans)
+        jsonObject.addProperty(TAG_ENV, env)
+
+        return jsonObject.toString()
+    }
+
+    // endregion
+
+    // region Internal
+
+    private fun sanitizeKeys(model: SpanEvent): SpanEvent {
+        val newUserObject = sanitizeUserAttributes(model.meta.usr)
+        val newMetricsObject = sanitizeMetrics(model.metrics)
+        return model.copy(meta = model.meta.copy(usr = newUserObject), metrics = newMetricsObject)
+    }
+
+    private fun sanitizeUserAttributes(usr: SpanEvent.Usr): SpanEvent.Usr {
```

Comment:
do you really need to sanitize here ? These are internal spans so I am not sure we need this. WDYT ?

Reply:
I am not sure, I copy them from `feature-traces` in case that we might need it, I can remove it since it's not useful